lib/screens/qpi_page.dart

```diff
@@ -1,15 +1,113 @@
+import 'package:admu_student_app/models/year.dart';
 import 'package:flutter/material.dart';
+import 'package:google_fonts/google_fonts.dart';
 
 class QPIPage extends StatefulWidget {
   @override
   _QPIPageState createState() => _QPIPageState();
 }
 
 class _QPIPageState extends State<QPIPage> {
+  List<YearDropDown> _years = <YearDropDown>[
+    YearDropDown(
+        header: "Year 1",
+        body: "FIRST SEMESTER"
+    ),
+
+    YearDropDown(
+        header: "Year 2",
+        body: "FIRST SEMESTER"
+    ),
+
+    YearDropDown(
+        header: "Year 3",
+        body: "FIRST SEMESTER"
+    ),
+  ];
   @override
   Widget build(BuildContext context) {
     return Container(
-      child: Text('This is the QPI Page'),
+      child: ListView(
+        children: [
+          ExpansionPanelList(
+            expansionCallback: (int index, bool isExpanded){
+              setState(() {
+                _years[index].isExpanded = !_years[index].isExpanded;
+              });
+            },
+
+            children:
+            _years.map((YearDropDown year){
+              return ExpansionPanel(
```

Comment:
ideally, this `ExpansionPanel` should be in a separate file `lib/widgets/qpi/filename.dart`, and then import it in this file

it can look like this:
```dart
// in qpi_page.dart
// ...
Widget build(BuildContext context) {
  // ... other stuff like text, cumulative qpi, etc.
  ListBuilder
    // ...
    return YearDropdown();
```

tho i haven't tried the `ListView` widget yet, only `ListView.builder`

Reply:
can you also briefly explain how this works?

Reply:
Sorry for the very late reply, I made the dropdown component in a separate file now, as well as including the input fields and color pickers. 
To explain how the dropdown works, there's a class called YearDropDown that has 3 String variables, 1 header and 2 bodies.
![image](https://user-images.githubusercontent.com/72382027/116868732-85a18300-ac39-11eb-8ab4-7acc9ffa7f62.png)
I then made a list of YearDropDowns:
![image](https://user-images.githubusercontent.com/72382027/116868789-a2d65180-ac39-11eb-854b-87c0ac2a1357.png)
Then created an ExpansionPanelList Widget using the Strings got from YearDropDown.
Regarding your draft, what should I do with it to merge with the component I made?
If there's any way to improve this code please let me know.

Reply:
oh i was referring to the `ListView` since i haven't used that one before; only `ListView.builder` so i dont know the differences between them.

supposedly, the QPI dropdown is the `ExpansionPanel` and not a list view:
```
// in qpi page

Column
  Text
  QPIView - cumulative qpi
  Text - qpi overview
  ListView.builder
    child:
      YearDropdown
```

something like that. think of one `widget` or `component` in the dev sprints as a singular button that you want to create over and over again across several files.
my goal here was that you guys in the front-end, won't have to copy and paste the same code for a single widget over and over and over again because that is tiring. and what if suddenly we have to change the border radius?

but don't worry, i can help out in creating the widgets for you guys ^^

regarding my draft, i'll be the one fixing the conflicts and hopefully by tomorrow, it's all in the develop branch and you guys can start working. thanks!

Reply:
it's also the reason why i told the designers to switch to figma components

another example: the semester buttons. you see these buttons **at least** seven times in the add qpi page, and it's also included in the add schedule page.